c7n/resources/cloudfront.py

```diff
@@ -181,6 +183,66 @@ def process(self, resources, event=None):
         return results
 
 
+@Distribution.filter_registry.register('check-s3-origin')
+class CheckS3Origin(Filter):
+    """Check for existence of S3 bucket referenced by Cloudfront, and verify ownership.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: cfront-test
+                resource: distribution
+                filters:
+                - type: check-s3-origin
+                  accounts_from:
+                    url: *accounts-list
+                    expr: accounts-list.canonical_id
+    """
+
+    schema = type_schema(
+        'check-s3-origin',
```

Comment:
I would rename this to origin, with 's3' being a type as cloudfront supports both s3 and http endpoints and whitelisting http endpoints shouldn't be that much of an extension to this policy